clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2818,15 +2818,14 @@ void CodeGenModule::SetCommonAttributes(GlobalDecl GD, llvm::GlobalValue *GV) {
     addUsedOrCompilerUsedGlobal(GV);
 
   if (getLangOpts().SYCLIsDevice) {
-    // Add internal device_global variables to llvm.compiler.used array to
-    // prevent early optimizations from removing these variables from the
-    // module.
+    // Add internal device_global variables to llvm.used array to prevent early
```

Comment:
How does this help? Just curious. Thanks

Reply:
I don't necessarily agree on using `llvm.used` or `llvm.compiler.used` but @AaronBallman suggested that we change uses of `llvm.compiler.used` to `llvm.used` [here](https://github.com/intel/llvm/pull/15148#discussion_r1736013305).